src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final String DEFAULT_COLOR = "White";
+    private static final double MAX_CIRCLE_RADIUS = 10;
+    private static final int RANDOM_INDEX = 6;
+    private final Random random;
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier() {
+        this.random = new Random();
+        this.colorSupplier = new ColorSupplier();
+    }
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(RANDOM_INDEX);
+        String color = colorSupplier.getRandomColor();
+        return switch (index) {
+            case 0 -> createRandomSquare(color);
+            case 1 -> createRandomRectangle(color);
+            case 2 -> createRandomRightTriangle(color);
+            case 3 -> createRandomCircle(color);
+            case 4 -> createRandomIsoscelesTrapezoid(color);
+            default -> getDefaultFigure();
```

Comment:
```suggestion
            default -> createRandomIsoscelesTrapezoid(color);
```
or throw an exception in default, this case just shouldn't happen